src/components/Intercom/Chat.js

```diff
@@ -0,0 +1,86 @@
+import React, { useEffect, useState } from "react";
+import useDocusaurusContext from "@docusaurus/useDocusaurusContext";
+import { Button } from "./Button";
+
+/**
+ * Wait for the global variable window.Intercom and the load it
+ * @param {object} data
+ */
+const waitForIntercomGlobal = (data) => {
+  if (window.Intercom) {
+    window.intercomSettings = data;
+    window.Intercom("boot");
+    window.Intercom("show");
+  } else {
+    setTimeout(() => {
+      waitForIntercomGlobal(data);
+    }, 50);
+  }
+};
+
+/**
+ * Load script
+ */
+const loadScript = ({ appId }) => {
+  return new Promise(function (resolve, reject) {
+    var script = document.createElement("script");
+    script.src = `https://widget.intercom.io/widget/${appId}`;
+    script.addEventListener("load", function () {
+      resolve();
+    });
+    script.addEventListener("error", function (e) {
+      reject(e);
+    });
+    document.body.appendChild(script);
+  });
+};
+
+export const Chat = () => {
+  const context = useDocusaurusContext();
+  const {
+    siteConfig: {
+      themeConfig: {
+        intercom: { appId },
+      },
+    },
+  } = context;
```

Comment:
```suggestion
  const appId = context.siteConfig?.themeConfig?.intercom?.appId;
```

Does docusaurus support this kind of syntax? If we gonna remove the whole `intercom` from the config - the build/start will fail completely. Using optional chaining will protect us from it.